Engine/main/update.cpp

```diff
@@ -81,48 +82,48 @@ int do_movelist_move(short*mlnum,int*xx,int*yy) {
 
     int adjAmnt = 3;
     // 2.70: if the X permove is also <=1, don't do the skipping
-    if (((xpermove & 0xffff0000) == 0xffff0000) ||
-        ((xpermove & 0xffff0000) == 0x00000000))
+    if ((trunc(xpermove) == -1) ||
+        (trunc(xpermove) == 0) )
       adjAmnt = 2;
 
     // 2.61 RC1: correct this to work with > -1 as well as < 1
-    if (ypermove == 0) { }
+    if (fabs(ypermove) < delta) { }
     // Y per move is < 1, so finish the move
-    else if ((ypermove & 0xffff0000) == 0)
+    else if (trunc(ypermove) == 0)
       targety -= adjAmnt;
     // Y per move is -1 exactly, don't snap to finish
-    else if (ypermove == 0xffff0000) { }
+    else if (fabs(ypermove - 1.0) < delta) { }
```

Comment:
This is not correct, the condition will succeed for values in range (+1.0 - delta) to (+1.0 + delta), but not for -1.0, as original code assumes.

Perhaps fabs is not even necessary here, you may do ```if ((ypermove >= -1.0 - delta) && (ypermove <= -1.0 + delta))```